lib/cext/preprocess.rb

```diff
@@ -115,11 +115,24 @@ def self.patch(file, contents, directory)
   end
 
   if __FILE__ == $0
-    puts "#line 1 \"#{ARGF.filename}\""
+    require 'stringio'
+    output_io = StringIO.new
+
+    file_name = ARGF.filename
+    original_content = File.read(file_name)
+    content = patch(file_name, original_content.clone, Dir.pwd)
```

Comment:
Why is there a defensive copy here? `patch` doesn't modify its argument, does it?

Reply:
right, it indeed does not